### ISSUE TITLE
token transfer command from the address without tokens

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#101]
I've tested `token transfer DTNewSpecv2_newl mgLyx4nNSPjcimazmo5U2Z8khUiRZWq5hn 1` command for the dPoD token.
My main address was `mtYvCVBtEayA5y6szGSrSgGwHQfe44Bgh2` and had no dPoD tokens on it, however the transaction has been run without any issue. The `txid` was `6b61ba823d9cf5998af8551d19d94c9adc3db38b11f176adced3ff3f2f3757c9`.
Of course I waited for a long time to see the dPoD tokens being transferred :-) until I haven't got that I'm on the wrong address.
Maybe we should prevent this happening because people would spend fees for nothing.

I've tested `token transfer ATTokenNewSpec2_local mtMGkB2mVxzPNZCWHruRmG1SmsSJBtyCUD 1` and `token transfer fb93cce7aceb9f7fda228bc0c0c2eca8c56c09c1d846a04bd6a59cae2a895974 mtMGkB2mVxzPNZCWHruRmG1SmsSJBtyCUD 1` setting an address without PoB tokens, `mgLyx4nNSPjcimazmo5U2Z8khUiRZWq5hn` as main and got the following error message:
```
  File "~/.local/bin/pacli", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/__main__.py", line 479, in main
    fire.Fire({
  File "~/.local/lib/python3.12/site-packages/fire/core.py", line 143, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/fire/core.py", line 477, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
                                ^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/fire/core.py", line 693, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/extended_classes.py", line 1307, in transfer
    return ei.run_command(self.__transfer, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/extended_interface.py", line 33, in run_command
    result = c(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/extended_classes.py", line 1330, in __transfer
    return eu.advanced_card_transfer(deck,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/extended_utils.py", line 382, in advanced_card_transfer
    issue_tx = pa.card_transfer(provider=provider,
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pypeerassets/__main__.py", line 288, in card_transfer
    unsigned_tx = make_raw_transaction(network=provider.network,
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pypeerassets/transactions.py", line 285, in make_raw_transaction
    return MutableTransaction(
           ^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pypeerassets/transactions.py", line 151, in __init__
    super().__init__(version, ins, outs, locktime, network=network, timestamp=timestamp)
  File "~/.local/lib/python3.12/site-packages/pypeerassets/transactions.py", line 56, in __init__
    if txid != self.txid and txid is not None:
               ^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/btcpy/structs/transaction.py", line 512, in txid
    return self.hash()
           ^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/btcpy/lib/types.py", line 28, in wrapper
    return method(self)
           ^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/btcpy/structs/transaction.py", line 507, in hash
    stream << self
    ~~~~~~~^^~~~~~
  File "~/.local/lib/python3.12/site-packages/btcpy/lib/parsing.py", line 373, in __lshift__
    self.body += other.serialize()
                 ^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/btcpy/lib/types.py", line 28, in wrapper
    return method(self)
           ^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pypeerassets/transactions.py", line 113, in serialize
    result << bytearray(chain.from_iterable(txout.serialize() for txout in self.outs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pypeerassets/transactions.py", line 113, in <genexpr>
    result << bytearray(chain.from_iterable(txout.serialize() for txout in self.outs))
                                            ^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/btcpy/lib/types.py", line 26, in wrapper
    object.__setattr__(self, cache, method(self))
                                    ^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/btcpy/structs/transaction.py", line 303, in serialize
    result << self.value.to_bytes(8, 'little')
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OverflowError: can't convert negative int to unsigned
```
I've got the same message while I've tried to transfer the brand new ATToken from the same address `mgLyx4nNSPjcimazmo5U2Z8khUiRZWq5hn` (without that type of tokens) set as main.
The exact commands that provides that error were: 
- `token transfer 7a2ae406ddf44ddb17532d4888fdc14573f52749445e9e014075c9f83cbe556f another_fresh 1` 
- `token transfer 7a2ae406ddf44ddb17532d4888fdc14573f52749445e9e014075c9f83cbe556f mwLKewF56R8Rje2tAuQRJzDxS9XC1a1q23 1`
- ` token transfer ATTokenNewSpec another_fresh 1`
and
- `token transfer  ATTokenNewSpec  mwLKewF56R8Rje2tAuQRJzDxS9XC1a1q23 1`.

**Comment by d5000** [comment]
Implemented the balance check in commit 4b3bea0. It can be overridden with `--nocheck/-n`, because it's significantly slower than without checking.

I have still to check if I can reproduce the error message you reported.

**Comment by buhtignew** [comment]
> I have still to check if I can reproduce the error message you reported. I was unsuccessful until now, having tried the tokens (Deck IDs) you mentioned.

After you've installed your commit https://github.com/slimcoin-project/pacli/commit/4b3bea00c00e40e2f1ec4a0cd8f37655d26f356a the error is gone. It was appearing on the address that was without that type of tokens. However now I'm not able to reproduce it any more, even with the `-n` flag.  

I think we can close this issue.
